### ISSUE TITLE
🐛 `options.propertyAssignment` in `lint/style/noParameterAssign` misunderstand `assignment`

### ISSUE THREAD

**Issue by mimorisuzuko** [biomejs/biome#8254]
### Environment information

```block
CLI:
  Version:                      2.3.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.9.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.6.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

Rules:

```json
"noParameterAssign": {
  "level": "error",
  "options": { "propertyAssignment": "deny" }
}
```

Output:

```javascript
function f(a) {
	const b = { c: 0 };

	b.c = a; // Assigning to a property of a function parameter is confusing.
	b.c = 1;
}
```

### Expected result

`b.c = a` do not throw any error because `a` is not assigned.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
